### PR TITLE
Update aws-sdk to version 3

### PR DESCRIPTION
**Pull request by kse-clearhaus**

### CODE REVIEW

**Comment by kse-clearhaus** [comment]
@ct-clearhaus when this is merged, can you update the [gem](https://rubygems.org/gems/dynamodb-mutex/versions/0.0.3)?

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -19,28 +19,38 @@
...
       end
     end
-    sleep(1)
-  end
 
-  config.after(:suite) do
-    FakeDynamo::Storage.instance.shutdown
-    dynamo_thread.exit if dynamo_thread
-    FileUtils.rm('test.fdb', force: true)
+    raise 'DynamoDB did not start in time' if loop_count >= loop_max_wait / sleep_interval
```
Comment (ct-clearhaus) [inline spec/spec_helper.rb]: What about something as simple as this?

```ruby
[0.1, 0.5, 1, 5, 10].map do |sleep_interval|
  Aws::DynamoDB::Client.new.list_tables rescue (sleep sleep_interval; false)
end.any? || fail('DynamoDB did not start in time')
```

Notice that it'll implicitly break 🙂

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -19,28 +19,38 @@
...
   end
 
+  config.after(:all) do
+    Process.kill('INT', dynamo_pid) if dynamo_pid
+    Process.waitpid(dynamo_pid)
```
Comment (ct-clearhaus) [inline spec/spec_helper.rb]: Should they both be cased by `if dynamo_pid`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    environment:
+      AWS_REGION: eu-west-1
+      AWS_ACCESS_KEY_ID: xxx
+      AWS_SECRET_ACCESS_KEY: xxx
+    steps:
+      - checkout
+      - run:
+          name: Install Java
+          command: |
+            export DEBIAN_FRONTEND=noninteractive
```
[the remark this answers is not in the document]
Comment (kse-clearhaus) [reply .circleci/config.yml]: Not right now :) I'll remove it.

File: spec/spec_helper.rb (hunk 1 above)
Comment (kse-clearhaus) [reply spec/spec_helper.rb]: `#golfing` :1st_place_medal: 

Will it implicitly break? Wont `[..].map <block>` evaluate the block for each value? What am I missing?

Comment (ct-clearhaus) [reply spec/spec_helper.rb]: Darn, I'm missing a `.lazy`:

```ruby
[0.1, 0.5, 1, 5, 10].lazy.map do |sleep_interval|
```

Comment (kse-clearhaus) [reply spec/spec_helper.rb]: Pretty sweet! :+1:

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,46 +1,68 @@
 require 'rubygems'
 require 'bundler/setup'
 require 'rspec'
-require 'fake_dynamo'
-require 'aws-sdk'
 
-Aws.config.update({
+require 'English'
```
Comment (ct-clearhaus) [inline spec/spec_helper.rb]: Not used any more — says @kse-clearhaus 🙂

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -1,46 +1,68 @@
...
+DYNAMODB_JAR_DIR = 'resources'.freeze
+DYNAMODB_TMP_PATH = '/tmp/dynamodb_local_latest.tar.gz'.freeze
+
+def fetch_amazon_dynamodb_local
+  Dir.mkdir(DYNAMODB_JAR_DIR) unless Dir.exist?(DYNAMODB_JAR_DIR)
+
+  return if File.exist?("./#{DYNAMODB_JAR_DIR}/DynamoDBLocal.jar")
+
+  raise "Error downloading #{AMAZON_LOCAL_DYNAMODB_URL}" unless \
+    system("wget #{AMAZON_LOCAL_DYNAMODB_URL} -O #{DYNAMODB_TMP_PATH} -q")
```
Comment (ct-clearhaus) [inline spec/spec_helper.rb]: Warning! Subjective bike-shed ahead!

I think I like something like this better:
```ruby
system("wget #{AMAZON_LOCAL_DYNAMODB_URL} -O #{DYNAMODB_TMP_PATH} -q") or raise "Error downloading #{AMAZON_LOCAL_DYNAMODB_URL}"
```
or
```ruby
x = system("wget #{AMAZON_LOCAL_DYNAMODB_URL} -O #{DYNAMODB_TMP_PATH} -q")
raise "Error downloading #{AMAZON_LOCAL_DYNAMODB_URL}" unless x
```
or ... dunno.
I think the thing I don't like is that "the primary thing" (the side effect `system()`) is "kinda hidden away" in the end of "the line".

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -1,46 +1,64 @@
...
+
+  raise "Error unpacking #{DYNAMODB_TMP_PATH}" unless \
+    system("tar -xf #{DYNAMODB_TMP_PATH} -C #{DYNAMODB_JAR_DIR}")
```
Comment (ct-clearhaus) [inline spec/spec_helper.rb]: `tar -xf` vs. `tar xf`?

File: spec/spec_helper.rb (hunk 5 above)
Comment (ct-clearhaus) [reply spec/spec_helper.rb]: Feel free to ignore this!

File: spec/spec_helper.rb (hunk 6 above)
Comment (kse-clearhaus) [reply spec/spec_helper.rb]: I don't have a (really) strong opinion on this, but for me it's a question of consistency.
I like my CLI arguments to be GNU style and for that reason add the dash by default.

File: spec/spec_helper.rb (hunk 5 above)
Comment (kse-clearhaus) [reply spec/spec_helper.rb]: I prefer the second option and completely agree with make the `system()` the primary command.

**Pull request merged**